presto-main/src/test/java/com/facebook/presto/operator/scalar/TestStringFunctions.java

```diff
@@ -123,6 +123,32 @@ public void testConcat()
         assertNotSupported("CONCAT(" + Joiner.on(", ").join(nCopies(255, "'1'")) + ")", "Too many arguments for string concatenation");
     }
 
+    @Test
+    public void testFormat()
+    {
+        String.format("%s", 1);
+        assertInvalidFunction("FORMAT('')", "There must be two or more arguments");
+        assertFunction("FORMAT('hello %s', 'world')", VARCHAR, "hello world");
+        assertFunction("FORMAT('a=%s, b=%s, c=%s', cast(1 as varchar), cast(2 as varchar), cast(3 as varchar))", VARCHAR, "a=1, b=2, c=3");
+        assertFunction("FORMAT('', '')", VARCHAR, "");
+        assertFunction("FORMAT('what', '')", VARCHAR, "what");
+        assertFunction("FORMAT('', 'what')", VARCHAR, "");
+        assertFunction("FORMAT(FORMAT('this %s %s', 'is'), 'cool')", VARCHAR, "this is cool");
+
+        // Test varbinary arguments
+        assertFunction("from_utf8(FORMAT(to_utf8('hello %s'), to_utf8('world')))", VARCHAR, "hello world");
+        assertFunction("from_utf8(FORMAT(to_utf8(''), to_utf8('')))", VARCHAR, "");
+
+        // Test concat for non-ASCII
+        assertFunction("FORMAT('hello na\u00EFve %s', 'world')", VARCHAR, "hello na\u00EFve world");
+        assertFunction("FORMAT('\uD801\uDC2D %s', 'end')", VARCHAR, "\uD801\uDC2D end");
+        assertFunction("FORMAT('\uD801\uDC2D%s%s', 'end', '\uD801\uDC2D')", VARCHAR, "\uD801\uDC2Dend\uD801\uDC2D");
```

Comment:
you can use actual letters directly, without Java unicode escapes. https://github.com/prestodb/presto/blob/master/presto-postgresql/src/test/java/com/facebook/presto/plugin/postgresql/TestPostgreSqlTypeMapping.java#L111